src/node.rs

```diff
@@ -0,0 +1,84 @@
+use squalid::return_default_if_false;
+use tree_sitter_grep::tree_sitter::{Node, TreeCursor};
+
+pub trait NodeExt<'a> {
```

Comment:
I'd started adding some ergonomic methods on `tree_sitter::Node` in the ESLint-rules plugin repo but made sense to "hoist" some of them here

I think in general ergonomics is a pretty big deal for the plugin-/rule-authoring experience so having a bunch of convenience methods for typical eg AST-traversal stuff (vs some of the `tree_sitter` crate API's being not bad but a little bit boilerplate-y-er) feels right to me